### PR TITLE
added set_progress_handler func

### PR DESCRIPTION
**Pull request by quaxsze**
### Description

Add bridge to set_progress_handler feature of sqlite python driver.

Fixes: #

### CODE REVIEW

File: aiosqlite/tests/smoke.py [hunk 1]
```diff
@@ -224,3 +224,13 @@ def __str__(self):
...
+
+    async def test_set_progress_handler(self):
+        """Assert that after setting a progress handler returning 1, DB operations are aborted"""
+        async with aiosqlite.connect(TEST_DB) as db:
+            await db.set_progress_handler(lambda: 1, 1)
+            with self.assertRaises(OperationalError) as context:
+                await db.execute(
+                    "create table test_progress_handler (i integer primary key asc, k integer)"
+                )
+                self.assertTrue(isinstance(context.exception, OperationalError))
```
Comment (jreese) [inline aiosqlite/tests/smoke.py]: I suspect this line/assertion will never be reached/tested, if the `db.execute(...)` is supposed to trigger an exception?

File: aiosqlite/core.py [hunk 2]
```diff
@@ -283,6 +283,9 @@ def total_changes(self) -> int:
     async def load_extension(self, path: str):
         await self._execute(self._conn.load_extension, path)  # type: ignore
 
+    async def set_progress_handler(self, handler: Callable, n: int) -> None:
```
Comment (jreese) [inline aiosqlite/core.py]: Looking at the sqlite3 docs, this would be better annotated as `handler: Callable[[], Optional[int]]`.

Comment (quaxsze) [reply aiosqlite/core.py]: Done

File: aiosqlite/tests/smoke.py (hunk 1 above)
Comment (quaxsze) [reply aiosqlite/tests/smoke.py]: Yes you are right I removed it

**Comment by jreese** [comment]
Cheers!

**Pull request merged**